pkg/controller/controller_binding.go

```diff
@@ -104,18 +103,17 @@ func (c *controller) reconcileServiceInstanceCredential(binding *v1alpha1.Servic
 		return err
 	}
 
-	// Determine whether the checksum has been invalidated by a change to the
-	// object.  If the binding's checksum matches the calculated checksum,
-	// there is no work to do.
+	// Determine whether there is a new generation of the object. If the binding's
+	// generation does not match the reconciled generation, then there is a new
+	// generation, indicating that changes have been made to the binding's spec.
 	//
 	// We only do this if the deletion timestamp is nil, because the deletion
 	// timestamp changes the object's state in a way that we must reconcile,
-	// but does not affect the checksum.
-	if binding.Status.Checksum != nil && binding.DeletionTimestamp == nil {
-		bindingChecksum := checksum.ServiceInstanceCredentialSpecChecksum(binding.Spec)
-		if bindingChecksum == *binding.Status.Checksum {
+	// but does not affect the generation.
+	if binding.Status.ReconciledGeneration != 0 && binding.DeletionTimestamp == nil {
```

Comment:
does it matter if ReconciledGeneration is zero?  Seems like we should only care about the deletion timestamp, no?